src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs

```diff
@@ -1201,14 +1221,14 @@ public void ToolsVersionOverrideCausesToolsetRedirect()
 
             ProjectInstance[] instances = null;
             SolutionFile solution = SolutionFile_Tests.ParseSolutionHelper(solutionFileContents);
-            BuildEventContext buildEventContext = new BuildEventContext(0, 0, 0, 0);
+            var buildEventContext = new BuildEventContext(0, 0, BuildEventContext.InvalidProjectContextId, 0);
```

Comment:
nit: you can extract this out in a field that gets instantiated in the constructor. This way you only have to define it once.